src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -937,8 +937,22 @@ public GHLabel createLabel(String name, String color) throws IOException {
     /**
      * Lists all the users who have starred this repo.
      */
-    public PagedIterable<GHUser> listStargazers() {
-        return listUsers("stargazers");
+    public PagedIterable<GHStargazer> listStargazers() {
+        return new PagedIterable<GHStargazer>() {
+            @Override
+            public PagedIterator<GHStargazer> _iterator(int pageSize) {
+                Requester requester = root.retrieve();
+                requester.setHeader("Accept", "application/vnd.github.v3.star+json");
```

Comment:
Added back the old version for compatibility and also listExtendedStargazers to return the new version. In addition added Javadoc to link both methods :)